src/framework/theme/components/search/search.component.ts

```diff
@@ -320,6 +329,10 @@ export class NbSearchComponent implements OnInit, OnDestroy {
     this.hideSearch();
   }
 
+  searchOnInput(term: string) {
+    this.searchService.submitSearch(term, this.tag);
```

Comment:
`input` event actually more like change than submit, so I don't think we should trigger submit here. It would be better to also add `searchInput` and `onSearchInput` methods to `NbSearchService`.